packages/itwinui-react/src/core/ThemeProvider/ThemeProvider.tsx

```diff
@@ -188,7 +188,7 @@ export const ThemeProvider = React.forwardRef((props, forwardedRef) => {
             <Root
               theme={theme}
               themeOptions={themeOptions}
-              ref={useMergedRefs(forwardedRef, setRootElement)}
+              ref={useMergedRefs(forwardedRef, setRootElement, useIuiDebugRef)}
```

Comment:
Correct, there will be a single warning for all versions on the page detected in 3 seconds.

The only time there will be separate warnings is when another version is loaded later (after the 3 seconds). I think that's a feature, not a bug, since we want to continue detecting all versions.